examples/functions-google-oauth/README.md

```diff
@@ -0,0 +1,50 @@
+# Gatsby Functions + Google OAuth Example
+
+This example illustrates logging in with Google 0Auth via Gatsby Functions and defining a [private cilent route behind this authentication](https://www.gatsbyjs.com/tutorial/authentication-tutorial/). The primary focus is on the Functions aspect, not the client side routing/authentication. [This article](https://soshace.com/react-user-login-authentication-using-usecontext-and-usereducer/#crayon-60c920bddc105062139412) has a good example for setting up the client side auth provider you could layer on top of this example.
+
+For a full walkthrough of setting up this example, including the Google Cloud OAuth setup, [follow this blog post]().
+
+![Image of Google Auth Gatsby App](./localhost_8000.png)
+
+## 🚀 Quick start
+
+1.  **Set up Google OAuth Client.**
+
+    This sample uses google oAuth client to authenticate into the Google inbox api. Obtain an oAuth credential from the Google Cloud [API & Services Credentails Page](https://console.cloud.google.com/apis/credentials?)
+
+    - Generate an Oauth Client ID. You can follow the steps highlighted in [Account Authentication - Service Account](https://theoephraim.github.io/node-google-spreadsheet/#/getting-started/authentication?id=oauth). Make sure to set up at least one **Authorized redirect URI** to be `http://localhost:8000`
+
+2.  **Add the Google Oauth Credentials to your `.env.development` file**
+
+    There are **2** environment variable you'll need to add to your project:
+
+    - `GOOGLE_CLIENT_ID`: Fetch the ClientId from the newly created OAuth 2.0 Client ID. Yuo can find it in the [Google Credentials Console](https://console.cloud.google.com/apis/credentials)
+    - `GOOGLE_CLIENT_SECRET`: Fetch the Client Secret from the newly created OAuth 2.0 Client ID. Yuo can find it in the [Google Credentials Console](https://console.cloud.google.com/apis/credentials)
+
+    You'll also want to add these as environment variables when deploying to Gatsby Cloud.
+
+3.  **Start developing.**
+
+    To get started, run `yarn` to add all necessary packages.
```

Comment:
Unfortunately the already merged examples used `yarn` in their examples. This should be fixed in a follow-up.

For this PR we can do it right before the merge. So any instance of `yarn` should be replaced with `npm`